### PR TITLE
Add more annotation customizations

### PR DESCRIPTION
**Pull request by jianghaolu**

### CODE REVIEW

File: customization-base/src/main/java/com/azure/autorest/customization/PropertyCustomization.java [hunk 1]
```diff
@@ -73,6 +73,108 @@ public PropertyCustomization rename(String newName) {
...
+        return this;
+    }
+
+    /**
+     * Remove an annotation from the property.
+     *
+     * @param annotation the annotation to remove from the property. The leading @ can be omitted.
+     * @return the current property customization for chaining
+     */
+    public PropertyCustomization removeAnnotation(String annotation) {
```
Comment (srnagar) [inline customization-base/src/main/java/com/azure/autorest/customization/PropertyCustomization.java:127]: Will this remove the annotation from the constructor too if the property had an annotation that was also used in the ctor args?

```java
    @JsonProperty(value = "length", required = true)
    private float length;

    @JsonCreator
    public Fish(@JsonProperty(value = "length", required = true) float length) {
        this.length = length;
    }
```

**Comment by srnagar** [comment]
For this change, do you also want to update the pom version for the customization library as this needs to be released to Maven.

File: customization-base/src/main/java/com/azure/autorest/customization/PropertyCustomization.java (hunk 1 above)
Comment (jianghaolu) [reply customization-base/src/main/java/com/azure/autorest/customization/PropertyCustomization.java:127]: The user will have to manually remove the annotation on the cotr. There may be future convenience customizations to "make a property optional".

**Comment by jianghaolu** [comment]
I have another small fix coming - I'll bump the pom version there.

**Pull request merged**